st2common/st2common/util/casts.py

```diff
@@ -62,6 +62,15 @@ def _cast_number(x):
 
 
 def _cast_string(x):
+    if x is None:
+        # Preserve None as-is
+        return x
```

Comment:
This is the actual fix for None type handling and issue @jjm encountered.

Rest of the changes are related user-friendliness improvements.